client/src/action_creators.js

```diff
@@ -147,9 +148,50 @@ export function fetchBit(bitId) {
   }
 }
 
-export function fetchBits(sort = SORT_DATE) {
+export function fetchBits({ sort, offset, limit } = {}) {
+  return function(dispatch, getState) {
+    return fetchBitsApi({
+      sort: sort || getState().getIn(['sorting', 'currentSort']),
+      offset: offset || getState().get('offset'),
+      pageSize: limit || getState().get('pageSize'),
+      dispatch: dispatch
+    });
+  }
+}
+
+export function fetchNextPage() {
+  return function(dispatch, getState) {
+    var offset = getState().get('offset') + getState().get('pageSize');
+    dispatch(clearBits());
```

Comment:
is this doing some kind of request state magic?

Reply:
Sort of yeah. fetchNextPage is [Thunk middleware][thunk], which basically lets us intercept Redux actions (used to change state) and do more interesting things. clearBits() is a Redux action that gets picked up by `reducer.js` to clear the `bits` field in the state, and then we fetch more Bits.

[thunk]: https://github.com/reduxjs/redux-thunk